### DISCUSSION TITLE
Create A  class for often used Runtime names (similar to vocabulary?)

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1265] [Dev]
Like mentioned here https://github.com/apache/streampipes/issues/1040#issuecomment-1426806856

What do you think about this: 

Is it possible to create and use RUNTIME_names that are used more often, like the vocabulary list, globally somewhere?
Instead of define it in each processor class it is stored like Geo.LAT vocabulary  in a separate class.

```
...
private static final String TEMPERATUR_RUNTIME="temperature"
...
        .outputStrategy(
            OutputStrategies.append(
                PrimitivePropertyBuilder
                    .create(Datatypes.String, TEMPERATUR_RUNTIME)
                    .domainProperty(Geo.LAT)
                    .build()
            )
...
```

It is stored in a Class Runtime
and is used like
```
        .outputStrategy(
            OutputStrategies.append(
                PrimitivePropertyBuilder
...
 ->                   .create(Datatypes.String, RUNTIME.TEMPERATUR)
...
                    .build()
            )
```

The advantage of this approach would be,
that the runtime name e.g. with the keep + updateField method, the correct runtime name can be selected more easily.

`event.addField(TEMPERATUR_RUNTIME_NAME, temperatur);`

`event.addField(RUNTIME.TEMPERATUR, temperatur);`

**Comment by bossenti** [comment]
Sounds like a good idea.
This could also help to streamline the naming of parameters among the several adapters and processing elements.
So +1 from my side :)

**Comment by flomickl** [comment]
Let's discuss it further
Should it be implemented like vocabulary?
Would it be necessary to create different categories for the runtime names?
If yes, what categories should we create?

https://github.com/apache/streampipes/tree/dev/streampipes-vocabulary/src/main/java/org/apache/streampipes/vocabulary

**Reply by bossenti** [reply]
How about integrating it with the vocabulary and by doing this restructuring the present vocabulary a bit?
From the first sight, an enum for runtime names is also somehow a vocabulary for me

What do you think?

**Reply by flomickl** [reply]
Do you mean in each vocabulary (topic) group add enums with default runtime names?
Sounds like a good plan. 
You add this also in #1373  :)

**Reply by bossenti** [reply]
Initially, I thought more about one runtime name enum but it might be a good idea to group them by domain 👍
